runtime/tests/checker/dictionary_test.go

```diff
@@ -42,11 +42,32 @@ func TestCheckIncompleteDictionaryType(t *testing.T) {
 
 	require.NoError(t, err)
 
-	assert.IsType(t,
+	assert.Equal(t,
 		&sema.DictionaryType{
 			KeyType:   sema.IntType,
 			ValueType: sema.InvalidType,
 		},
 		RequireGlobalValue(t, checker.Elaboration, "dict"),
 	)
 }
+
+func TestCheckMetaKeyType(t *testing.T) {
+
+	t.Parallel()
+
+	checker, err := ParseAndCheck(t,
+		`
+		let dict: {Type: String} = {Type<Int>(): "a"}
+        `,
+	)
+
+	require.NoError(t, err)
+
+	assert.Equal(t,
+		&sema.DictionaryType{
+			KeyType:   sema.MetaType,
+			ValueType: sema.StringType,
+		},
+		RequireGlobalValue(t, checker.Elaboration, "dict"),
```

Comment:
This assertion is not very effective, as it just asserts the explicitly stated type annotation for the variable declaration. Maybe remove the type annotation for `dict`.